Applying YAPF formatting to Telescope

mtlynch opened the pull request:
This change updates the style of all Python files in the directory by applying [YAPF](https://github.com/google/yapf) to them so that they adhere to the M-Lab style guide (i.e. the Google style guide). This does not make it 100% compliant because there are style rules that cannot be automated, but it does automate a lot of uninteresting style fixes for us.

It also adds a check to the Travis config so that builds will fail if they diverge from the YAPF-formatted code (keeps us from diverging from correct style). That commit (21dc74e) is the only human-written code change in this PR. All other changes are the result of running the code through YAPF with:

```
yapf --in-place --recursive --style=google ./
```

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/m-lab/telescope/71)
<!-- Reviewable:end -->

mtlynch commented:
To see what it looks like when the style diverges from YAPF-produces format see this build, where I purposely added a line that was > 80 characters long:

https://travis-ci.org/mtlynch/telescope/builds/82207383

collina commented on telescope/external.py:
```diff
@@ -54,96 +55,100 @@ class BigQueryCommunicationError(Exception):
-    credentials = storage.get()
+        flow = flow_from_clientsecrets(
+            os.path.join(os.path.dirname(__file__),
+                         'resources/client_secrets.json'),
```
When do we indent to the starting paran, like here, vs. line 115 where we start indented from the line and not the parans.

mtlynch replied:
If the first parameter is on the same line as the left paren, we indent subsequent parameters to align with the first parameter. If the first param starts on the next line, we add an indent from the previous line:

```
do_something('/somepath/foo/bar/baz/longfilename.txt',
             '/otherpath/baz/boo/bar/longfilename.txt')

do_other_thing(
    '/superlongpaththatdoesntevenfitonthesamelineasfuncname/baz.txt',
    'cats.txt')
```

collina commented on telescope/external.py:
```diff
@@ -54,96 +55,100 @@ class BigQueryCommunicationError(Exception):
+    def collect_results(self, job_id):
+        """Wait until a job is complete and gather all results.
 
     Args:
```
Is this aligned right?

mtlynch replied:
Ah, good catch. It's getting this wrong. It looks like a bug in YAPF. I'll investigate.

collina commented on telescope/external.py:
```diff
@@ -151,25 +156,26 @@ def collect_results(self, job_id):
+        page_token = None
+
+        while is_first_chunk or page_token:
+            results_response = self._wait_for_results_chunk(job_id, page_token)
```
These few lines need wrap?

mtlynch replied:
Either way works. I don't see anything wrong with YAPF's choice, do you?

collina commented on telescope/external.py:
```diff
@@ -182,32 +188,35 @@ def _wait_for_results_chunk(self, job_id, page_token):
+                if retries_remaining > 0:
+                    sleep_seconds = 10
+                    logging.warning(
+                        ('Failed to communicate with BigQuery to retrieve '
```
wrap

mtlynch replied:
Same as the case above. We can probably do line breaks better with the changed indent.

collina commented:
Review status: 0 of 21 files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

<sup>**[telescope/external.py, line 279 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Ymi5b6BoZgo2GaqUD)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/external.py#L279)):</sup>
Not sure I understand this spacing?
<!-- Reviewable comment -K-Ymi5c98QAeJrjuxiW:-1344556724 -->

---

<sup>**[telescope/external.py, line 297 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Ymo5E6tsnvqRMCrOC)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/external.py#L297)):</sup>
Same here?
<!-- Reviewable comment -K-Ymo5E6tsnvqRMCrOD:1662569722 -->

---

<sup>**[telescope/filters.py, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Yn5VZBUGNd-XL0tWA)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/filters.py#L61)):</sup>
Think the method documentation is often misaligned.
<!-- Reviewable comment -K-Yn5VZBUGNd-XL0tWB:428161620 -->

---

<sup>**[telescope/iptranslation.py, line 206 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YndyoDF2MbJZBpX6Y)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/iptranslation.py#L206)):</sup>
Is this right?
<!-- Reviewable comment -K-YndyoDF2MbJZBpX6Z:-649655340 -->

---

<sup>**[telescope/selector.py, line 174 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YoMDhPj5nbumNhnvT)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/selector.py#L174)):</sup>
Is this right?
<!-- Reviewable comment -K-YoMDieScP5ehoAblK:-649655340 -->

---

<sup>**[telescope/telescope.py, line 123 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YofZxOu5a1ctjoy3A)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/telescope.py#L123)):</sup>
Is this right?
<!-- Reviewable comment -K-YofZxOu5a1ctjoy3B:-649655340 -->

---

<sup>**[telescope/utils.py, line 40 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Yp1wX7pl-mSQ2mBad)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/utils.py#L40)):</sup>
Extraneous linebreak?
<!-- Reviewable comment -K-Yp1wX7pl-mSQ2mBae:1301113929 -->

---

<sup>**[tests/test_external.py, line 146 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YpH75EXKk5b2jouYF)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/tests/test_external.py#L146)):</sup>
Seems like weird formatting to me?
<!-- Reviewable comment -K-YpH75EXKk5b2jouYG:1021326573 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/telescope/71)
<!-- Sent from Reviewable.io -->

mtlynch commented:
Review status: 0 of 21 files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

<sup>**[telescope/external.py, line 153 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Ti1E5RNd2x1GWcXjL-r2-153)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/external.py#L153)):</sup>
Confirmed bug in YAPF: https://github.com/google/yapf/issues/60
<!-- Reviewable comment -K-Z2WYqLtH4bhDI1vu2:1169151479 -->

---

<sup>**[telescope/external.py, line 279 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Ymi5b6BoZgo2GaqUD)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/external.py#L279)):</sup>
YAPF was breaking up the line because it was > 80 chars, but it doesn't seem to care much about keeping list comprehensions on a single line. Fixed manually.
<!-- Reviewable comment -K-Z2tWw4-6K47WFRrix:-1056839887 -->

---

<sup>**[telescope/external.py, line 297 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Ymo5E6tsnvqRMCrOC)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/external.py#L297)):</sup>
Looks like a bug in YAPF. Fixed manually and filed: https://github.com/google/yapf/issues/189
<!-- Reviewable comment -K-Z6XokCvF538ILH9LO:1124651350 -->

---

<sup>**[telescope/filters.py, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Yn5VZBUGNd-XL0tWA)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/filters.py#L61)):</sup>
Fixed manually
<!-- Reviewable comment -K-Z6g9EanOdZL53497Y:-1366201500 -->

---

<sup>**[telescope/iptranslation.py, line 206 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YndyoDF2MbJZBpX6Y)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/iptranslation.py#L206)):</sup>
It is not, fixed manually.
<!-- Reviewable comment -K-Z2QzBewypR2HhFFqg:-142901042 -->

---

<sup>**[telescope/selector.py, line 174 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YoMDhPj5nbumNhnvT)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/selector.py#L174)):</sup>
I think it's syntactically valid, but not an ideal line break. Fixed manually.
<!-- Reviewable comment -K-Z2Knm58gk9tjMG_iG:573160865 -->

---

<sup>**[telescope/telescope.py, line 123 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YofZxOu5a1ctjoy3A)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/telescope.py#L123)):</sup>
Looks right to me. First param is on the following line with one indent.
<!-- Reviewable comment -K-Z1nu6TadgNuVbB6xz:-180822700 -->

---

<sup>**[telescope/utils.py, line 40 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-Yp1wX7pl-mSQ2mBad)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/utils.py#L40)):</sup>
This is correct. Top level functions should be preceded by two blank lines: https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Blank_Lines#Blank_Lines
<!-- Reviewable comment -K-Z6pgndZ9CyOc9MRWT:1361521968 -->

---

<sup>**[tests/test_external.py, line 146 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YpH75EXKk5b2jouYF)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/tests/test_external.py#L146)):</sup>
Agreed. Unfortunately YAPF won't play nice with this one, so I had to ungracefully disable YAPF for this line.
<!-- Reviewable comment -K-Z8yhGFm9F_LiiaKj7:88050338 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/telescope/71)
<!-- Sent from Reviewable.io -->

collina commented:
Review status: 0 of 21 files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

<sup>**[telescope/telescope.py, line 123 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YofZxOu5a1ctjoy3A)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/telescope.py#L123)):</sup>
You wouldn't move len down instead?
<!-- Reviewable comment -K-ZbVZz4vWfBZg1gLxV:-168554347 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/telescope/71)
<!-- Sent from Reviewable.io -->

mtlynch commented:
Review status: 0 of 21 files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

<sup>**[telescope/telescope.py, line 123 \[r2\]](https://reviewable.io:443/reviews/m-lab/telescope/71#-K-YofZxOu5a1ctjoy3A)** ([raw file](https://github.com/m-lab/telescope/blob/96770ea62abeabedcb41ad51d373e03d8fbc068a/telescope/telescope.py#L123)):</sup>
Oh, I didn't realize that's what you meant. Either look fine for me, but if you have a preference for len to not be split up like that, I'm happy to change it. Fixed.
<!-- Reviewable comment -K-Zk0_0Vzf-qtq6ticq:1667910305 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/telescope/71)
<!-- Sent from Reviewable.io -->

mtlynch commented:
LGTM from @collina 

Thanks for the review!
